src/lib/protocols/kerberos.c

```diff
@@ -402,6 +617,15 @@ void ndpi_search_kerberos(struct ndpi_detection_module_struct *ndpi_struct,
 #endif
 
 	      koffsetp = koffset + 4;
+	      if (krb_parse(ndpi_struct, flow, koffsetp) == 0)
+	      {
+	        NDPI_LOG_DBG(ndpi_struct,
+	                     "[TGS-REP][s/dport: %u/%u][Kerberos Hostname,Domain,Username][%s,%s,%s]\n",
+	                     sport, dport, flow->protos.kerberos.hostname, flow->protos.kerberos.domain,
+	                     flow->protos.kerberos.username);
+	        flow->extra_packets_func = NULL;
+	        return;
+	      }
```

Comment:
Let's see if I got that right.
You wrote a new decoder for the TGS-REP message. If your code succeeds, dissection ends. But if your code fails, the existing code is triggered. Is that the expected behaviour?

Reply:
This is a more or less quick fix for #1492. I was pretty much scared about unforseen consequences. That is why the old code is still triggered.